scr/sysbox

```diff
@@ -30,49 +30,163 @@ function retry() {
   false
 }
 
-# kill any running sysbox processes
-sysmgr_pids=$(pidof sysbox-mgr)
-sysfs_pids=$(pidof sysbox-fs)
-
-for pid in $sysmgr_pids; do
-   kill $pid
-done
-
-for pid in $sysfs_pids; do
-   kill $pid
-done
-
-# Increase default inotify resources to meet sys container's demands.
-sysctl -w fs.inotify.max_queued_events=1048576
-sysctl -w fs.inotify.max_user_watches=1048576
-sysctl -w fs.inotify.max_user_instances=1048576
-
-# Increase default keyring resources to meet sys container demands.
-# For a k8s cluster:
-# keys = 35 + (workers * 23) + (2 * pods)
-# maxbytes = 20 bytes * maxkeys
-sysctl -w kernel.keys.maxkeys=20000
-sysctl -w kernel.keys.maxbytes=$((20*20000))
+# Ensure that kernel-modules expected by system-level apps (running within sys
+# containers) are properly loaded.
+load_required_modules() {
 
-#
-# Start Sysbox components
-#
-mkdir -p /var/lib/sysboxfs
-sysbox-mgr --log /dev/stdout > /var/log/sysbox-mgr.log 2>&1 &
+  # Sysbox requires 'configfs' module to ensure proper operation of containarized
+  # apps requiring access to kernel's config file (e.g. kubeadm).
+  if ! modprobe configfs &> /dev/null; then
+      echo "Could not load configfs kernel module. Exiting ..."
+      return 1
+  fi
+
+  return 0
+}
+
+# Returns linux distro running in the system.
+function get_host_distro() {
+
+  local distro=$(cat /etc/os-release | awk -F"=" '/^ID=/ {print $2}' | tr -d '"')
+
+  echo $distro
+}
+
+# Ensures unprivileged user-ns are allowed.
+function setup_userns() {
+
+  local distro=$1
+
+  if [[ "${distro}" == "ubuntu" ]] ||
+     [[ "${distro}" == "debian" ]]; then
+      echo "1" > /proc/sys/kernel/unprivileged_userns_clone
```

Comment:
In centos/fedora/redhat: is there a file that performs the same function as " /proc/sys/kernel/unprivileged_userns_clone" ?

Reply:
Nope, there's no such a file, this is the file utilized by podman and others. I'll be updating sysbox-runc soon too.